src/Features/Core/Portable/Completion/Providers/AbstractEnumCompletionProvider.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Recommendations;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Completion.Providers
+{
+    internal abstract class AbstractEnumCompletionProvider : AbstractSymbolCompletionProvider
+    {
+        private static readonly CompletionItemRules s_rules = CompletionItemRules.Default.WithMatchPriority(MatchPriority.Preselect);
```

Comment:
how does this work given that the enum type itself will be preselected?

Reply:
I don't know (copied from the original VB provider). Should I use `CompletionItemRules.Default` instead? That seems to be the right choice given that the enum type is preselected.

Reply:
i think the two languages have different behaviors.  sounds like C# should have different rules here.

Reply:
I played around with this setting in the IDE and at the moment the behavior between C# and VB is the same: The first enum value is pre-selected (even if the enum type is listed above). I think, that is a sensible behavior. I don't have written any tests for that, thought.
These tests should cover the interaction between the SymbolCompletionProvider (offering the Enum type), this provider, pre selection, and item ordering. I will add these tests later.

Reply:
> I played around with this setting in the IDE and at the moment the behavior between C# and VB is the same: The first enum value is pre-selected (even if the enum type is listed above). I think, that is a sensible behavior. I don't have written any tests for that, thought.

i don't think tihs is reasonable as this breaks c# muscle memory.  i.e. in c# i type: 

`Color color = <tab><dot>` and i expect to be given a list of members of `Color`.  With your apporach, the `tab` will complete to the first member which is not how we've worked for the last 15 years.  Please change htis :)

Reply:
> Color color = <tab><dot> and i expect to be given a list of members of Color. With your apporach, the tab will complete to the first member which is not how we've worked for the last 15 years. Please change this :)

I totally agree. 

> I played around with this setting in the IDE and at the moment the behavior between C# and VB is the same: The first enum value is pre-selected (even if the enum type is listed above). I think, that is a sensible behavior. I don't have written any tests for that, thought.

I have to take back that statement. I have written some [C# tests now](https://github.com/dotnet/roslyn/pull/49632/files#diff-31303c421816ee9001acbb872ec31e6ef3d701cd04eba6f2a41d02e49d70ae21R2324), that show, that the enum type is still preselected (as it should).

I think in my previous investigation I committed the selection before and so next time the committed enum value was preselected (that is at least what I think happened).

I could still make the `s_rule` provider dependent to be absolutely sure. What do you think?

Reply:
> I could still make the s_rules provider dependent to be absolutely sure. What do you think?

Yup. I would do that

> I think in my previous investigation I committed the selection before and so next time the committed enum value was preselected (that is at least what I think happened).

Can we have a test that verifies that that doesn't happen?  If we commit an enum value, and then invoke completion again, we should still preselect the enum, not the enum value. Thanks!

Reply:
> Can we have a test 

I added one in c758cb2

> I could still make the s_rules provider dependent to be absolutely sure.

Did that in 9d0f198

If you look at the test, the enum value is preselected in the second invocation of the completion list. This is true no matter if `GetCompletionItemRules` is overridden or not (Default vs. `WithMatchPriority(MatchPriority.Preselect)`. The test before and after 9d0f198 passes.

I think this is the standard behavior of the completion list: Pre-select the last committed item.